### PR TITLE
Add 3.2.1.2 test case

### PR DESCRIPTION
**Pull request by nimanthag**
Signed-off-by: nimanthag <nimanthag@wso2.com>

## User stories
API developer should be able to assign an already defined scope to at least one of the valid resources.

## Developer testing record
screen shot 2019-02-01 at 12 15 55 pm

### CODE REVIEW

File: product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java [hunk 1]
```diff
@@ -136,13 +182,42 @@ private void deleteRoles() throws APIManagementException {
         }
     }
 
+    private void createAndPublishAPI()throws APIManagerIntegrationTestException{
```
Comment (mushthaq33) [inline product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java]: Formatting issue

File: product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java [hunk 2]
```diff
@@ -136,13 +182,42 @@ private void deleteRoles() throws APIManagementException {
...
+        HttpResponse serviceResponse = apiPublisher.addAPI(apiRequest);
+        verifyResponse(serviceResponse);
+        HttpResponse serviceResponseGetApi = apiPublisher.getAPI(apiName, SUPER_USER);
+
+        APILifeCycleStateRequest updateRequest = new APILifeCycleStateRequest(apiName, SUPER_USER,
+                APILifeCycleState.PUBLISHED);
+        apiPublisher.changeAPILifeCycleStatus(updateRequest);
+    }
+
+    public static String readFromFile(String file_name) throws IOException {
```
Comment (praminda) [inline product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java]: I believe we moved this to a common util class

Comment (praminda) [reply product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java]: method is implemented in ScenarioTestUtils shall we use it here?

File: product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java (hunk 1 above)
Comment (praminda) [inline product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java]: Will you be able to move this method to the same util class I've mentioned below. We can create the API request in the test case and pass it to the util method

Comment (rmsamitha) [inline product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java]: Shall we use this common method for this.
https://github.com/wso2/product-apim/blob/cd655777f335d12a56f15e377637c8ebf0f08e7a/product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIPublisherRestClient.java#L928

Comment (nimanthag) [reply product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java]: Done

Comment (nimanthag) [reply product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesTestCases.java]: This method removed and used a common method.

**Pull request merged**